packages/components/src/unit-control/index.js

```diff
@@ -159,11 +159,18 @@ function UnitControl(
 		/>
 	) : null;
 
-	// Get current unit type
-	const activeUnit = units.find( ( option ) => option.value === unit );
-
-	// Extract the step
-	const unitStep = activeUnit?.step ? activeUnit.step : 1;
+	let step = props.step;
+	// if no step prop has been passed
+	if ( ! step ) {
+		// Get current unit type
+		let activeUnit = units.find( ( option ) => option.value === unit );
+		// if unit wasn't found, choose the first option
+		if ( ! activeUnit && units.length > 0 ) {
+			activeUnit = units[ 0 ];
+		}
```

Comment:
If we can't find a matching unit, it doesn't really make sense to grab the first one. We should instead leave early with `step = 1`.

Reply:
I'm inclined to agree, however to explain my logic - when I was doing some testing (putting in an empty value), the unit was not detected (no active unit).

In this scenario, it looks like the first unit (from the list of units) is selected (visually) due to the nature of a select even when it has no value.  This just meant, even in those edge cases, we were still respecting the step associated with the unit type - but I guess that is more or less catering for a unlikely/unwanted use case?

Reply:
Removed this in the latest commit.